pyproject.toml

```diff
@@ -49,3 +53,73 @@ unfixable = ["F401"]
 [tool.ruff.mccabe]
 # Unlike Flake8, default to a complexity level of 10.
 max-complexity = 10
+
+[tool.pdm]
+version = { source = "file", path = "autogen/version.py" }
+
+[project]
+name = "pyautogen"
+dynamic = ["version"]
+description = "Enabling Next-Gen LLM Applications via Multi-Agent Conversation Framework"
+readme = "README.md"
+authors = [
+    {name = "AutoGen", email = "auto-gen@outlook.com"},
+]
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+]
+requires-python = ">=3.9, <3.12"
+dependencies = [
+    "diskcache",
+    "flaml",
+    "openai~=1.3",
+    "python-dotenv",
+    "termcolor",
+    "tiktoken",
+    "openai>=1.3",
+    "pydantic>=1.10,<3",
+    "docker>=7.0.0",
+]
+
+[project.urls]
+Homepage = "https://github.com/microsoft/autogen"
+
+[tool.pdm.dev-dependencies]
+test = [
```

Comment:
This way, we lose `test` target and the following will not work:
``` sh
pip install -e .[test]
```

Why not adding `test` target to `[project.optional-dependencies]` section?

Reply:
"You can have several groups of development only dependencies. Unlike optional-dependencies, they won't appear in the package distribution metadata such as PKG-INFO or METADATA. The package index won't be aware of these dependencies."

"Local directories and VCS dependencies can be installed in [editable mode](https://pip.pypa.io/en/stable/cli/pip_install/#editable-installs). If you are familiar with pip, it is just like pip install -e <package>. **Editable packages are allowed only in development dependencies**:"

"The same group name MUST NOT appear in both [tool.pdm.dev-dependencies] and [project.optional-dependencies]."

Reply:
I know, but is this what we want?